pkg/astnormalization/implicit_extend_root_operation.go

```diff
@@ -0,0 +1,85 @@
+package astnormalization
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+/*
+type Query {...}
+type Mutation {...}
+type Subscription {...}
+
+will be,
+
+extend type Query {...}
+extend type Mutation {...}
+extend type Subscription {...}
+
+this also works if root types are defined in schema{...} with other names.
+root types are left unmodified if they have no fields, directives or implements any interface.
+*/
+type implicitExtendRootOperationVisitor struct {
+	operation               *ast.Document
+	schemaRootOperationName []string
```

Comment:
This can be even more simplified!
Have a look at the AST struct here: https://github.com/TykTechnologies/graphql-go-tools/blob/master/pkg/ast/ast.go#L55

The Index already contains the names of Query, Mutation and Subscription. So we could remove some logic from here :+1: